test/services/apimap-field-builder.test.js

```diff
@@ -78,6 +78,74 @@ describe('services > apimap-field-builder', () => {
     });
   });
 
+  describe('on a column with allowNull set to true', () => {
+    const fieldDefinitions = {
+      allowNullString: {
+        type: Sequelize.DataTypes.STRING,
+        allowNull: true,
+      },
+    };
+
+    it('should not be set as required', () => {
+      expect.assertions(1);
+      const { allowNullString } = initializeField(fieldDefinitions);
+      expect(allowNullString.isRequired).toBeUndefined();
+    });
+  });
+
+  describe('on a column with allowNull set to false', () => {
+    const fieldDefinitions = {
+      disallowNullString: {
+        type: Sequelize.DataTypes.STRING,
+        allowNull: false,
+      },
+    };
+
+    it('should be set as required', () => {
+      expect.assertions(1);
+      const { disallowNullString } = initializeField(fieldDefinitions);
+      expect(disallowNullString.isRequired).toStrictEqual(true);
+    });
+  });
+
+  describe('on a column with allowNull not set', () => {
+    const fieldDefinitions = {
+      someString: {
+        type: Sequelize.DataTypes.STRING,
+      },
+    };
+
+    it('should not be set as required', () => {
+      expect.assertions(1);
+      const { someString } = initializeField(fieldDefinitions);
+      expect(someString.isRequired).toBeUndefined();
+    });
+  });
+
+  describe('on a column that does not allowNull values & is a primary key', () => {
```

Comment:
[Code Organisation] "on a column that does not allowNull values" is already un existing `describe` so this case should be a sub-describe.